### ISSUE TITLE
Kernel BUG at metadata_raw.c:254! during metadata corruption test

### ISSUE THREAD

**Issue by Deixx** [Open-CAS/open-cas-linux#1044]
## Description

Kernel BUG occurred after metadata corruption. 

## Logs

```
[141343.886652] kernel BUG at /root/open-cas-linux/modules/cas_cache/src/ocf/metadata/metadata_raw.c:254!
[141343.901397] invalid opcode: 0000 [#1] SMP PTI
[141343.910793] CPU: 18 PID: 2915779 Comm: cas_mngt_1 Kdump: loaded Tainted: G          IOE    --------- -  - 4.18.0-305.el8.x86_64 #1
[141343.951438] RIP: 0010:_raw_ram_load_all+0xfd/0x120 [cas_cache]
[141343.963928] Code: 5d 41 5c 41 5d 41 5e 41 5f e9 2f 24 fe ff 5b 5d 41 5c 41 5d 41 5e 41 5f c3 31 c0 45 85 c0 0f 95 c0 48 85 c0 0f 84 2e ff ff ff <0f> 0b 5b 4c 89 f7 5d 4c 89 f8 41 5c be bc bd f0 ff 41 5d 41 5e 41
[141343.998058] RSP: 0018:ffffb81c790e7e60 EFLAGS: 00010202
[141344.010777] RAX: 0000000000000001 RBX: 00000000ffffff70 RCX: ffffb81c47151028
[141344.025901] RDX: ffffffffc0c4a7e0 RSI: ffffb81c46de9150 RDI: ffffb81c7e4f5000
[141344.041199] RBP: ffff9f960969f600 R08: 0000000000200000 R09: ffffffffc0c87760
[141344.056656] R10: 0000000000000020 R11: 0000000000000000 R12: ffff9fa03d4f2280
[141344.072216] R13: 0000000000000000 R14: ffff9fa03d4f22c0 R15: ffff9f960969f600
[141344.087890] FS:  0000000000000000(0000) GS:ffff9fa09f400000(0000) knlGS:0000000000000000
[141344.104854] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[141344.119335] CR2: 00007fa2f8f64004 CR3: 000000079fa10003 CR4: 00000000001706e0
[141344.135551] Call Trace:
[141344.146444]  _ocf_pipeline_run_step+0xad/0xd0 [cas_cache]
[141344.160848]  ocf_queue_run+0x23/0x40 [cas_cache]
[141344.174418]  _cas_io_queue_thread+0x6a/0x100 [cas_cache]
[141344.188895]  ? finish_wait+0x80/0x80
[141344.201398]  ? cas_service_ioctl_ctrl+0x29f0/0x29f0 [cas_cache]
[141344.216755]  kthread+0x116/0x130
[141344.229049]  ? kthread_flush_work_fn+0x10/0x10
[141344.242867]  ret_from_fork+0x35/0x40
[141344.255676] Modules linked in: cas_cache(OE) cas_disk(OE) brd ext4 mbcache jbd2 nls_utf8 isofs cdrom nvme nvme_core xt_CHECKSUM ipt_MASQUERADE xt_conntrack dm_mod ipt_REJECT nf_reject_ipv4 nft_compat nft_counter nft_chain_nat nf_nat nf_conntrack nf_defrag_ipv6 nf_defrag_ipv4 nf_tables nfnetlink tun bridge stp llc loop rpcsec_gss_krb5 auth_rpcgss nfsv4 dns_resolver igb nfs lockd grace fscache sunrpc vfat fat intel_rapl_msr intel_rapl_common sb_edac x86_pkg_temp_thermal intel_powerclamp coretemp kvm_intel kvm iTCO_wdt iTCO_vendor_support irqbypass crct10dif_pclmul crc32_pclmul ftdi_sio ghash_clmulni_intel joydev lpc_ich i2c_i801 ipmi_si rapl ioatdma ipmi_devintf intel_cstate intel_uncore ipmi_msghandler mei_me pcspkr mei wmi acpi_pad acpi_power_meter ip_tables xfs libcrc32c sd_mod t10_pi sg drm_kms_helper ahci syscopyarea sysfillrect libahci sysimgblt fb_sys_fops crc32c_intel drm libata dca i2c_algo_bit fuse [last unloaded: cas_disk]
```

## Your Environment

* OpenCAS version (commit hash or tag): 21.06.4.0553.master
* Operating System: RHEL8.4
* Kernel version: 4.18.0-305.el8.x86_64

**Comment by karolinavelkaja** [comment]
Changes under review

**Comment by kmajzero** [comment]
Closed by merging ocf PR #654
